components/input/Input.js

```diff
@@ -4,26 +4,66 @@ import PropTypes from 'prop-types';
 import { generateUID } from '../../helpers/component';
 import useInput from './useInput';
 import ErrorZone from '../text/ErrorZone';
-import { omit } from 'proton-shared/lib/helpers/object';
 
-const Input = React.forwardRef((props, ref) => {
-    const { className, error, ...rest } = omit(props, ['onPressEnter']);
-    const { handlers, statusClasses, status } = useInput(props);
-    const [uid] = useState(generateUID('input'));
-    return (
-        <>
-            <input
-                className={`pm-field w100 ${className} ${statusClasses}`}
-                aria-invalid={error && status.isDirty}
-                aria-describedby={uid}
-                ref={ref}
-                {...rest}
-                {...handlers}
-            />
-            <ErrorZone id={uid}>{error && status.isDirty ? error : ''}</ErrorZone>
-        </>
-    );
-});
+const Input = React.forwardRef(
+    (
+        {
+            error,
+            autoComplete = 'off',
+            autoFocus,
+            className = '',
+            disabled,
+            id,
+            name,
+            onBlur,
+            onChange,
+            onFocus,
+            onKeyDown,
+            onKeyUp,
+            onPressEnter,
+            placeholder,
+            readOnly,
+            required,
+            type = 'text',
+            value
+        },
```

Comment:
I'd like to keep passing rest if possible, something like this:

```js
const Input = React.forwardRef(({ className = '', error, autoComplete = 'off', type = 'text', onPressEnter, ...rest }, ref) => {
    const { handlers, statusClasses, status } = useInput({ ...rest, onPressEnter });
    const [uid] = useState(generateUID('input'));
    return (
        <>
            <input
                className={`pm-field w100 ${className} ${statusClasses}`}
                aria-invalid={error && status.isDirty}
                aria-describedby={uid}
                ref={ref}
                autoComplete={autoComplete}
                type={type}
                {...rest}
                {...handlers}
            />
            <ErrorZone id={uid}>{error && status.isDirty ? error : ''}</ErrorZone>
        </>
    );
});
```
(and same for Select)

Do you see a reason that wouldn't work?

Reply:
I hesitated whether to write that, but I thought maybe it was a strange mix (some props types defined but the props not declared, some not defined nor declared, some defined and declared). But definitively it looks better code-wise, so if you prefer that, I'm all for it!